src/Microsoft.ML.Transforms/WrappedGcnTransformers.cs

```diff
@@ -0,0 +1,221 @@
+                    pairs.Add((inputNames[((OutPipelineColumn)outCol).Input], outputNames[outCol]));
+
+                return new LpNormalizer(env, pairs.ToArray(), _normKind, _subMean);
+            }
+        }
+
+        /// <include file='doc.xml' path='doc/members/member[@name="LpNormalize"]/*'/>
+        /// <param name="input">The column to apply to.</param>
+        /// <param name="normKind">Type of norm to use to normalize each sample.</param>
+        /// <param name="subMean">Subtract mean from each value before normalizing.</param>
+        public static Vector<float> LpNormalize(this Vector<float> input, NormalizerKind normKind = NormalizerKind.L2Norm, bool subMean = false) => new OutPipelineColumn(input, normKind, subMean);
+    }
+
+    /// <summary>
+    /// Extensions for statically typed GcNormalizer estimator.
+    /// </summary>
+    public static class GcNormalizerExtensions
+    {
+        private sealed class OutPipelineColumn : Vector<float>
+        {
+            public readonly Vector<float> Input;
+
+            public OutPipelineColumn(Vector<float> input, bool subMean, bool useStdDev, float scale)
+                : base(new Reconciler(subMean, useStdDev, scale), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler
+        {
+            private readonly bool _subMean;
+            private readonly bool _useStdDev;
+            private readonly float _scale;
+
+            public Reconciler(bool subMean, bool useStdDev, float scale)
+            {
+                _subMean = subMean;
+                _useStdDev = useStdDev;
+                _scale = scale;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                Contracts.Assert(toOutput.Length == 1);
+
+                var pairs = new List<(string input, string output)>();
+                foreach (var outCol in toOutput)
+                    pairs.Add((inputNames[((OutPipelineColumn)outCol).Input], outputNames[outCol]));
+
+                return new GcNormalizer(env, pairs.ToArray(), _subMean, _useStdDev, _scale);
+            }
+        }
+
+        /// <include file='doc.xml' path='doc/members/member[@name="LpNormalize"]/*'/>
+        /// <param name="input">The column to apply to.</param>
+        /// <param name="subMean">Subtract mean from each value before normalizing.</param>
+        /// <param name="useStdDev">Normalize by standard deviation rather than L2 norm.</param>
+        /// <param name="scale">Scale features by this value.</param>
+        public static Vector<float> GcNormalize(this Vector<float> input,
```

Comment:
>GcNormalize [](start = 36, length = 11)

While 'Lp' is fine (it's a mathematical notation L_p), 'Gc' is probably not. 'GlobalContrast' ? Does it even make sense to have both as extensions though? They seem to share a lot in common

Reply:
I think they have different set of parameters and mixing them into one would not look good. I have renamed `GcNormalize` to `GlobalContrastNormalize`.

---
In reply to: [219260214](https://github.com/dotnet/machinelearning/pull/961#discussion_r219260214) [](ancestors = 219260214)